### PR TITLE
feat(ui): events table

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Fetch and display the events in the logs tab.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the React logs tab for a machine.
- You should see a list of events (pagination etc. coming in future PRs).

## Fixes

Fixes: canonical-web-and-design/maas-squad#2217.

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineLogs/EventLogs/EventLogsTable/EventLogsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import eventSelectors from "app/store/event/selectors";
+import type { EventRecord } from "app/store/event/types";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine } from "app/store/machine/types";
+import type { RootState } from "app/store/root/types";
+
+type Props = { systemId: Machine["system_id"] };
+
+type NetworkRow = {
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineLogs/EventLogs/EventLogsTable/EventLogsTable.tsx:14]: Lies!

File: ui/src/scss/_patterns_icons.scss [hunk 2]
```diff
@@ -113,6 +113,24 @@
     background-size: 100% 100%;
   }
 
+
+  .p-icon--debug {
```
Comment (Caleb-Ellis) [inline ui/src/scss/_patterns_icons.scss:117]: Vanilla got a bunch of new icons recently. Do you think any of them could work here instead?

File: ui/src/scss/_patterns_icons.scss [hunk 3]
```diff
@@ -113,6 +113,24 @@
...
+    @extend %icon;
+    background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' xmlns='http://www.w3.org/2000/svg'%3E%3Ctitle%3Edebug icon 8@2x%3C/title%3E%3Cpath d='M11.673 12.994L14.68 16l1.313-1.316-3.004-3.005A5.501 5.501 0 0 0 8.5 3C5.46 3 3 5.463 3 8.5 3 11.54 5.46 14 8.5 14c1.18 0 2.276-.37 3.173-1.006zM4.25 8.5a4.25 4.25 0 1 1 8.5 0 4.25 4.25 0 0 1-8.5 0zM6 9h5v1H6V9zm0-2h5v1H6V7zM4.71 0C2.845 0 1.646.095.87.87.093 1.648 0 2.847 0 4.716v4.568c0 1.87.094 3.068.87 3.844.667.668 1.68.824 3.13.857v-1.002c-1.327-.044-2.075-.21-2.424-.56-.41-.41-.576-1.318-.576-3.14V4.716c0-1.82.167-2.727.576-3.137C1.986 1.168 2.892 1 4.71 1h4.577c1.82 0 2.726.17 3.135.578.348.35.514 1.097.558 2.422h1.006c-.033-1.45-.19-2.46-.857-3.13C12.352.096 11.153 0 9.286 0H4.71z' fill='gray' fill-rule='evenodd'/%3E%3C/svg%3E");
+    background-size: 100% 100%;
+  }
+
+  .p-icon--audit,
```
Comment (Caleb-Ellis) [inline ui/src/scss/_patterns_icons.scss:123]: I don't think we should be mapping the api to css classes. Could we handle this in the component instead, using existing Vanilla classes?

**Review by Caleb-Ellis — APPROVED** [review]
Just a few suggestions

File: ui/src/app/machines/views/MachineDetails/MachineLogs/EventLogs/EventLogsTable/EventLogsTable.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineLogs/EventLogs/EventLogsTable/EventLogsTable.tsx:14]: Hah, thanks

File: ui/src/app/machines/views/MachineDetails/MachineLogs/EventLogs/EventLogsTable/EventLogsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  );
+  const events = useSelector((state: RootState) =>
+    eventSelectors.getByNodeId(state, machine?.id)
+  );
+  if (!machine || !events) {
+    return <Spinner text="Loading..." />;
+  }
+
+  const rows = events
+    .sort(
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineLogs/EventLogs/EventLogsTable/EventLogsTable.tsx:54]: Yep, good idea.

File: ui/src/scss/_patterns_icons.scss (hunk 3 above)
Comment (huwshimi) [reply ui/src/scss/_patterns_icons.scss:123]: Yep, I've mapped these in the component.

File: ui/src/scss/_patterns_icons.scss (hunk 2 above)
Comment (huwshimi) [reply ui/src/scss/_patterns_icons.scss:117]: Yep, I'll use the debug one, it looks a bit different, but probably better than a custom icon.

**Pull request merged**